pkg/cloud/rgraph/rnode/backendservice/type_trait.go

```diff
@@ -43,136 +43,30 @@ func (*typeTrait) FieldTraits(v meta.Version) *api.FieldTraits {
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("Backends").AnySliceIndex().Pointer().Field("MaxRatePerInstance"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("Backends").AnySliceIndex().Pointer().Field("MaxUtilization"))
-
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LocalityLbPolicies"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LocalityLbPolicies").AnySliceIndex().Pointer().Field("CustomPolicy"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LocalityLbPolicies").AnySliceIndex().Pointer().Field("CustomPolicy").Pointer().Field("Data"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LocalityLbPolicies").AnySliceIndex().Pointer().Field("Policy"))
-
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LocalityLbPolicy"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LogConfig"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LogConfig").Pointer().Field("Enable"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LogConfig").Pointer().Field("OptionalFields"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("LogConfig").Pointer().Field("SampleRate"))
-
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("MaxStreamDuration"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("MaxStreamDuration").Pointer().Field("Nanos"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("MaxStreamDuration").Pointer().Field("Seconds"))
-
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("Metadatas"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("OutlierDetection"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("OutlierDetection").Pointer().Field("EnforcingConsecutiveErrors"))
-
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("SecuritySettings"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("SecuritySettings").Pointer().Field("AwsV4Authentication"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("SecuritySettings").Pointer().Field("AwsV4Authentication").Pointer().Field("AccessKeyId"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("SecuritySettings").Pointer().Field("AwsV4Authentication").Pointer().Field("AccessKeyVersion"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("SecuritySettings").Pointer().Field("ClientTlsPolicy"))
-	dt.AllowZeroValue(api.Path{}.Pointer().Field("SecuritySettings").Pointer().Field("SubjectAltNames"))
-
-	if v == meta.VersionBeta || v == meta.VersionAlpha {
-		dt.AllowZeroValue(api.Path{}.Pointer().Field("SecuritySettings").Pointer().Field("Authentication"))
-		dt.AllowZeroValue(api.Path{}.Pointer().Field("CircuitBreakers").Pointer().Field("ConnectTimeout"))
-		dt.AllowZeroValue(api.Path{}.Pointer().Field("CircuitBreakers").Pointer().Field("ConnectTimeout").Pointer().Field("Nanos"))
-		dt.AllowZeroValue(api.Path{}.Pointer().Field("CircuitBreakers").Pointer().Field("ConnectTimeout").Pointer().Field("Seconds"))
-		dt.AllowZeroValue(api.Path{}.Pointer().Field("Subsetting").Pointer().Field("SubsetSize"))
-
+	dt.Mandatory(api.Path{}.Pointer().Field("LoadBalancingScheme"))
```

Comment:
The mandatory fields are only those fields which are mandatory in reality in the GCE API?

Reply:
for BackendService I made fields mandatory when they are mandatory or have non zero default value (like Protocol and CompressionMode)

Reply:
SG. we clarified it during VC.